experiment/simulation/js/circuit.js

```diff
@@ -0,0 +1,210 @@
+// This function checks map when called
+
+function checkAndUpdate() {
+    // these variables are for pseudo nmos circuit
+    nmosNand = 0;
+    pmosNand = 0;
+    listOutput[0].voltage = 0;
+    // if any vdd is connected to any pmos store voltage
+    for (i = 0; i < listVdd.length; i++) {
+        for (j = 0; j < listPmos.length; j++) {
+            const r = listVdd[i].id.concat("$", listPmos[j].id);
+            if (connectionMap.has(r)) {
+                listPmos[j].voltage = 5;
+
+            }
+        }
+
+    }
+    // if any ground is connected to any pmos store voltage
+    for (i = 0; i < listGround.length; i++) {
+        for (j = 0; j < listPmos.length; j++) {
+            const r = listGround[i].id.concat("$", listPmos[j].id);
+            if (connectionMap.has(r)) {
+                listPmos[j].voltage = -5;
+
+            }
+        }
+
+    }
+    // if amy vdd is connected to nmos store that voltage
+    for (i = 0; i < listVdd.length; i++) {
+        for (j = 0; j < listNmos.length; j++) {
+            const r = listVdd[i].id.concat("$", listNmos[j].id);
+            if (connectionMap.has(r)) {
+                listNmos[j].voltage = 5;
+
+            }
+        }
+
+    }
+    for (i = 0; i < listGround.length; i++) {
+        for (j = 0; j < listNmos.length; j++) {
+            const r = listGround[i].id.concat("$", listNmos[j].id);
+            if (connectionMap.has(r)) {
+                listNmos[j].voltage = -5;
+
+            }
+        }
+
+    }
+    for (i = 0; i < listInput.length; i++) {
+        for (j = 0; j < listPmos.length; j++) {
+            const r = listInput[i].id.concat("$", listPmos[j].id);
+            if (connectionMap.has(r)) {
+                if (listInput[i].input === 0) {
+
+                    if (listPmos[j].voltage === 5) {
+                        listPmos[j].outvoltage = 5;
+                    }
+                    else {
+                        if (listPmos[j].voltage === 0) {
+                            listPmos[j].outvoltage = 9;
+                        }
+                        else {
+                            listPmos[j].outvoltage = -5;
+                        }
+
+                        //listPmos[j].outvoltage = 5;
+                    }
+                    console.log("Voltage stored in pmos");
```

Comment:
Remove the console log statements wherever present.

Reply:
Removed them